SNOW-764946: 3.0.2 release

sfc-gh-aling opened the pull request:
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-764946

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

sfc-gh-mkeller commented:
Don't forget to run https://github.com/snowflakedb/snowflake-connector-python/actions/workflows/create_req_files.yml

sfc-gh-aling commented:
@sfc-gh-mkeller 

added the corresponding three:

```
  - Fixed a bug of incorrect type hints of `SnowflakeCursor.fetch_arrow_all` and `SnowflakeCursor.fetchall`.
  - Improved logging to mask tokens in case of errors.
  - Fixed a bug where `snowflake.connector.util_text.split_statements` swallows the final line break in the case when there are no space between lines.
```

sfc-gh-mkeller reviewed (approved):
:ship: :ship:

sfc-gh-mheavin commented:
Release Notes:

Version 3.0.2 (March 21, 2023)
New Features
None.
Bug Fixes
Fixed a bug of incorrect type hints of SnowflakeCursor.fetch_arrow_all and SnowflakeCursor.fetchall.
Improved logging to mask tokens in case of errors.
Fixed a bug where snowflake.connector.util_text.split_statements swallowed the final line break in the case when there are no space between lines.

sfc-gh-aling commented on DESCRIPTION.md:
```diff
@@ -8,10 +8,13 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 # Release Notes
 
-- v3.0.2(Unreleased)
+- v3.0.2(March 22, 2023)
```
```suggestion
- v3.0.2(March 21, 2023)
```

sfc-gh-aling commented on DESCRIPTION.md:
```diff
@@ -8,10 +8,13 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 # Release Notes
 
-- v3.0.2(Unreleased)
+- v3.0.2(March 21, 2023)
```
```suggestion
- v3.0.2(March 22, 2023)
```

sfc-gh-aling commented on DESCRIPTION.md:
```diff
@@ -8,9 +8,13 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 # Release Notes
 
-- v3.0.2(Unreleased)
+- v3.0.2(March 22, 2023)
```
```suggestion
- v3.0.2(March 23, 2023)
```

sfc-gh-aling commented on DESCRIPTION.md:
```diff
@@ -8,9 +8,13 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
+  - Fixed a bug of incorrect type hints of `SnowflakeCursor.fetch_arrow_all` and `SnowflakeCursor.fetchall`.
+  - Improved logging to mask tokens in case of errors.
+  - Fixed a bug where `snowflake.connector.util_text.split_statements` swallows the final line break in the case when there are no space between lines.
   - Fixed a bug where the `put` command on AWS raised `AttributeError` when the file size was larger than 200M.
```
```suggestion
```

sfc-gh-aling commented on DESCRIPTION.md:
```diff
@@ -8,10 +8,13 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
-  - Fixed a bug where the `put` command on AWS raised `AttributeError` when the file size was larger than 200M.
+  - Fixed a bug where the `put` command on AWS raised `AttributeError` when uploading file composed of multiple parts.
+  - Fixed a bug of incorrect type hints of `SnowflakeCursor.fetch_arrow_all` and `SnowflakeCursor.fetchall`.
+  - Improved logging to mask tokens in case of errors.
+  - Fixed a bug where `snowflake.connector.util_text.split_statements` swallows the final line break in the case when there are no space between lines.
```
```suggestion
  - Fixed a bug where `snowflake.connector.util_text.split_statements` swallows the final line break in the case when there are no space between lines.
  - Improved logging to mask tokens in case of errors.
```
